### PR TITLE
fix(errorcodes): add ErrorCodes: WorkspaceDatabaseUserTypeMustBeAdmin

### PR DESCRIPTION
**Pull request by CHLK**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:
add ErrorCodes: WorkspaceDatabaseUserTypeMustBeAdmin
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by yhilmare — APPROVED** [review]
lgtm

File: server/odc-core/src/main/java/com/oceanbase/odc/core/shared/constant/ErrorCodes.java [hunk 1]
```diff
@@ -305,6 +305,11 @@ public enum ErrorCodes implements ErrorCode {
      * version conflict
      */
     WorksheetEditVersionConflict,
+
+    /**
+     * workspace
+     */
+    WorkspaceDatabaseUserTypeMustBeAdmin,
```
Comment (yizhouxw) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/shared/constant/ErrorCodes.java:312]: we would better add a CloudErrorCodes in cloud code repository.
CloudErrorCodes also implements ErrorCode interface

**Review by yizhouxw — APPROVED** [review]
keep current in this sprint

File: server/odc-core/src/main/resources/i18n/ErrorMessages.properties [hunk 2]
```diff
@@ -197,4 +197,5 @@ com.oceanbase.odc.ErrorCodes.PartitionKeyDataTypeMismatch=The data type does not
 com.oceanbase.odc.ErrorCodes.TimeDataTypePrecisionMismatch=The time type precision does not match. The configured time interval precision is greater than the maximum precision of the partition key {0} data type.
 com.oceanbase.odc.ErrorCodes.DatabaseAccessDenied=Database access is denied because the current user does not have {0} permissions for the database.
 com.oceanbase.odc.ErrorCodes.ObQueryProfileNotSupported=Query profile is only available for OceanBase Database with versions equal to or higher than {0}.
-com.oceanbase.odc.ErrorCodes.WorksheetEditVersionConflict=Someone has just modified this file. Please refresh the page to get the latest version and continue editing.
\ No newline at end of file
+com.oceanbase.odc.ErrorCodes.WorksheetEditVersionConflict=Someone has just modified this file. Please refresh the page to get the latest version and continue editing.
```
Comment (JessieWuJiexi) [inline server/odc-core/src/main/resources/i18n/ErrorMessages.properties:200]: 这个工作簿需要修改为worksheet么？

Comment (CHLK) [reply server/odc-core/src/main/resources/i18n/ErrorMessages.properties:200]: okay, i'll make this change

**Pull request merged**